Removes Ansible reinstall in CI pipeline

luksi1 opened the pull request:
**What type of PR is this?**

> /kind cleanup

**What this PR does / why we need it**:

This removes the Ansible reinstall that is run on each job of the pipeline. These steps are already performed in the image. This should speed up the pipeline considerably.

**Which issue(s) this PR fixes**:

Fixes #10031 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note
[CI] Removes Ansible reinstall from build pipeline
```

luksi1 commented:
@MrFreezeex, I know you aren't assigned as a reviewer, but this work is in the same vain as the Dockerfile and pipleine work if you want to take a peak.

MrFreezeex reviewed (approved):
Sure, this make a lot of sense to me, thanks! :D 
/ok-to-test
/lgtm

MrFreezeex commented:
Actually it currently change something: when we test with the 2.11 requirements. We might just drop supporting two versions altogether, both are end of life anyway right now so I don't believe it should change much...

luksi1 commented:
> Actually it currently change something: when we test with the 2.11 requirements. We might just drop supporting the two requirements versions altogether and just keep one, both versions are end of life anyway right now so I don't believe it should change much...

Good catch @MrFreezeex! I missed the CI variable for `ANSIBLE_MAJOR_VERSION` and was under the presumption that we were simply re-installing Ansible, since the default values from the shell script were set to 2.12 as well as the `requirements.txt`.

I'm going to set this to draft because it certainly isn't ready-to-go as is, *but* I'd like to know how we should move forward with this, since speeding up, and slimming down, the test pipeline has value both from an administration maintenance point-of-view as well as giving developers faster feedback on their PRs. Either 1) we can agree that going forward, we'll only support one version of Ansible, get that work done, rebase this, and then simply merge this as-is or 2) we believe that using two versions is best for now and in that case, I'll alter this PR to include virtuell environments for each `requirements.txt` and then we should be ready-to-go with this. What are your thoughts?

It might even be nice to get some feedback as well from @floryut or @cristicalin, or some other maintainers, as this decision would change Kubespray's support matrix, moving from two Ansible versions down to one, if that's the consensus.

luksi1 commented:
@MrFreezeex, I just saw the changes in #10034 and it looks like that PR moves to a one Ansible version support model. If that is the case, let's rebase this after #10034 and we should be good to go.

MrFreezeex commented:
Hi @luksi1, now that we have only one supported version of Ansible do you have some time to rebase your PR :pray:?

MrFreezeex commented:
Could you remove the draft status as well :pray:?
/lgtm

luksi1 commented:
The draft status is now removed @MrFreezeex. I've also added `ansible-base` to the `requirements.txt` and ensured that it's a compatible version with ansible 7.6.0. 

Good work on getting us moved over to a single Ansible version!

MrFreezeex commented:
Hmmm the image doesn't build now :thinking:, guessing it's because of the `ansible-base` thing :thinking:

luksi1 commented:
You are right @MrFreezeex and it appears that `ansible-base` is deprecated **and** `ansible-core` is a dependency on `ansible`, so `ansible-base` has been wrapped into `ansible-core` and `ansible-core` has become a dependency on `ansible`, so we only need to install `ansible` and then let `ansible` handle the dependency matrix. You can read about it [here](https://www.ansible.com/blog/ansible-3.0.0-qa) and [here](https://blog.while-true-do.io/ansible-release-3-0-0/).

MrFreezeex commented:
> You are right @MrFreezeex and it appears that `ansible-base` is deprecated **and** `ansible-core` is a dependency on `ansible`, so `ansible-base` has been wrapped into `ansible-core` and `ansible-core` has become a dependency on `ansible`, so we only need to install `ansible` and then let `ansible` handle the dependency matrix. You can read about it [here](https://www.ansible.com/blog/ansible-3.0.0-qa) and [here](https://blog.while-true-do.io/ansible-release-3-0-0/).

Indeed I also double checked and ansible 7.6.0 has a dependency with ansible-core like `ansible-core~=2.14.6`. So seems fair indeed :+1: 
/lgtm

luksi1 commented:
@MrFreezeex, do we need do anything more to get this merged?
